schema/formats.go

```diff
@@ -46,22 +47,53 @@ func (f macFormatChecker) IsFormat(input string) bool {
 	return match
 }
 
+func matchYangIPv4AddressPattern(input string) bool {
+	res, err := regexp.MatchString(`^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])(%[\p{N}\p{L}]+)?`, input)
+
+	if err != nil {
+		return false
+	}
+	return res
+}
+
+func (f ipv4FormatChecker) IsFormat(input string) bool {
+	ip := net.ParseIP(input)
+	if !((ip != nil) && (ip.To4() != nil)) {
```

Comment:
for simplicity in code reading, I suggest helper functions + changing the `if` condition a little bit:

```Go
func isValid(ip net.IP) bool {
   return ip != nil
}

func isV4(ip net.IP) bool {
   return ip.To4() != nil
}

// in this function
if !isValid(ip) || !isV4(ip) {
   return false
}
```

Reply:
Good point. Looks simpler that way.